pkg/controllers/cloud/cloudprovideraccount_controller.go

```diff
@@ -83,8 +95,18 @@ func (r *CloudProviderAccountReconciler) Reconcile(ctx context.Context, req ctrl
 }
 
 func (r *CloudProviderAccountReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	var err error
 	r.accountProviderType = make(map[types.NamespacedName]common.ProviderType)
-	if err := ctrl.NewControllerManagedBy(mgr).For(&crdv1alpha1.CloudProviderAccount{}).Complete(r); err != nil {
+	// Client in controller requires reconciler for each object that are under watch. So to avoid reconciler and use only
+	// watch, that can be implemented by clientset.
+	if r.clientset, err = kubernetes.NewForConfig(ctrl.GetConfigOrDie()); err != nil {
+		r.Log.Error(err, "error creating client config")
+		return err
+	}
+	// Using GenerationChangedPredicate to allow CPA controller to receive CPA updates
+	// for all events except change in status.
+	if err := ctrl.NewControllerManagedBy(mgr).
```

Comment:
I have added this change in my PR. Its not required here.

Reply:
Okay.

Reply:
Solved.